src/sentry/api/endpoints/organization_events_trace.py

```diff
@@ -19,26 +21,27 @@ def is_root(item):
     return item["root"] == "1"
 
 
-def serialize_event(event, parent, is_root_event=False):
-    return {
-        "event_id": event["id"],
-        "span_id": event["trace.span"],
-        "transaction": event["transaction"],
-        "transaction.duration": event["transaction.duration"],
-        "project_id": event["project_id"],
-        "parent_event_id": parent,
-        # Avoid empty string for root events
-        "parent_span_id": event["trace.parent_span"] or None,
-        "is_root": is_root_event,
-    }
-
-
-class OrganizationEventsTraceLightEndpoint(OrganizationEventsEndpointBase):
+class OrganizationEventsTraceEndpointBase(OrganizationEventsEndpointBase):
     def has_feature(self, organization, request):
         return features.has(
             "organizations:trace-view-quick", organization, actor=request.user
         ) or features.has("organizations:trace-view-summary", organization, actor=request.user)
 
+    def serialize_event(self, event, parent, generation, is_root_event=False):
+        return {
+            "event_id": event["id"],
+            "span_id": event["trace.span"],
+            "transaction": event["transaction"],
+            "transaction.duration": event["transaction.duration"],
+            "project_id": event["project_id"],
+            "parent_event_id": parent,
+            # Avoid empty string for root events
+            "parent_span_id": event["trace.parent_span"] or None,
+            # TODO(wmak) remove once we switch over to generation
+            "is_root": is_root_event,
+            "generation": generation,
```

Comment:
Can we leave a comment that this can be `None` for the light trace because we don't always know what generation it is?

Reply:
:+1: I'll default the param to `None` too so its clearer that its possible